checker/func_cyclo_checker.go

```diff
@@ -0,0 +1,92 @@
+package checker
+
+import (
+	"fmt"
+	"go/ast"
+
+	"github.com/uber-go/mapdecode"
+)
+
+func init() {
+	must(Register("func_cyclo", NewFuncCycloChecker))
+}
+
+// FuncCycloConfig describes the configuration of a FuncCycloChecker.
+type FuncCycloConfig struct {
+
+	// Max is the maximum cyclomatic complexity of a func.
+	Max int `mapdecode:"max"`
+}
+
+// FuncCycloChecker checks that funcs are within specific cyclomatic complexity.
+type FuncCycloChecker struct {
+	max int
+}
+
+// NewFuncCycloChecker constructs a FuncCycloChecker.
+func NewFuncCycloChecker(configData interface{}) NodeChecker {
+	var config FuncCycloConfig
+	if err := mapdecode.Decode(&config, configData); err != nil {
+		return nil
+	}
+
+	return &FuncCycloChecker{
+		max: config.Max,
+	}
+}
+
+// Title implements the NodeChecker interface.
+func (c *FuncCycloChecker) Title() string {
+	return "Func Cyclo Complexity"
+}
+
+// Description implements the NodeChecker interface.
+func (c *FuncCycloChecker) Description() string {
+	return fmt.Sprintf(`The maximum cyclomatic complexity of a func is %d.`, c.max)
```

Comment:
Can you add some reference as to what is 'cyclomatic complexity'?

Reply:
Not yet, it will require some markup in the string for the URL and a parser to display it correctly in the web view. I added a TODO.